axum-connect/src/handler/handler_stream.rs

```diff
@@ -187,56 +137,14 @@ macro_rules! impl_handler {
 
                     let res = stream! {
                         while let Some(item) = res.next().await {
-                            let rpc_item = item.rpc_into_response();
-                            match rpc_item {
-                                Ok(rpc_item) => {
-                                    if binary {
-                                        let mut res = vec![0, 0, 0, 0, 0];
-                                        if let Err(e) = rpc_item.encode(&mut res) {
-                                            let e = RpcError::new(RpcErrorCode::Internal, e.to_string());
-                                            yield Result::<Vec<u8>, Infallible>::Ok(encode_error(&e, true));
-                                            break;
-                                        }
-                                        let size = ((res.len() - 5) as u32).to_be_bytes();
-                                        res[1..5].copy_from_slice(&size);
-                                        yield Ok(res);
-                                    } else {
-                                        let mut res = vec![0, 0, 0, 0, 0];
-                                        if let Err(e) = serde_json::to_writer(&mut res, &rpc_item) {
-                                            let e = RpcError::new(RpcErrorCode::Internal, e.to_string());
-                                            yield Ok(encode_error(&e, true));
-                                            break;
-                                        }
-                                        let size = ((res.len() - 5) as u32).to_be_bytes();
-                                        res[1..5].copy_from_slice(&size);
-                                        yield Ok(res);
-                                    }
-                                },
-                                Err(e) => {
-                                    yield Ok(encode_error(&e, binary));
-                                    break;
-                                }
-                            }
+                            yield Ok(item.rpc_into_response()?);
```

Comment:
I **really** dig not having this logic in the macro definition, nice work! It was a pain to write and debug as a macro. I would even be happy removing the commented-out impl above, as this macro now does very little apart from type coercion.